src/freak.jl

```diff
@@ -0,0 +1,105 @@
+type FREAK{S, T, O} <: DescriptorParams
+    pattern_scale::Float64
+    pattern_table::Array{Array{S, 1}, 1}
+    smoothing_table::Array{T, 1}
+    orientation_weights::Array{O, 1}
+end
+
+typealias OrientationPair Vector{Int}
+typealias SamplePair Vector{Float64}
+
+function FREAK(; pattern_scale::Float64 = 22.0)
+    pattern_table, smoothing_table = _freak_tables(pattern_scale)
+    orientation_weights = OrientationPair[]
+    for o in freak_orientation_sampling_pattern
+        offset_1 = pattern_table[1][o[1]]
+        offset_2 = pattern_table[1][o[2]]
+        dy, dx = offset_1 - offset_2
+        norm = (dx ^ 2 + dy ^ 2)
+        push!(orientation_weights, OrientationPair([round(Int, dy * 4096 / norm), round(Int, dx * 4096 / norm)]))
+    end
+    FREAK(pattern_scale, pattern_table, smoothing_table, orientation_weights)
+end
+
+function _freak_mean_intensity{T<:Gray}(int_img::AbstractArray{T, 2}, keypoint::Keypoint, offset::SamplePair, sigma::Float64)
+    y = keypoint[1] + offset[1]
+    x = keypoint[2] + offset[2]
+    if sigma < 0.5
+        tl = bilinear_interpolation(int_img, y - sigma, x - sigma)
+        tr = bilinear_interpolation(int_img, y - sigma, x + sigma)
+        bl = bilinear_interpolation(int_img, y + sigma, x)
+        br = bilinear_interpolation(int_img, y + sigma, x + sigma)
+        return (br + tl - tr - bl) / (4 * (sigma ^ 2))
```

Comment:
I would say the best would be to write out the formula for the mean intensity over the chosen region in terms of the original (pre-integral) image. You can of course calculate those original pixel intensities from the integral image.

This is another good reason to turn your integral image into a type: there's no sense in burying these calculations inside a method specific to FREAK, instead you want to make them available to anyone who might want to use them.

If you want to give it a stab but don't quite know how to start, just ask. Some of my packages like `IndirectArrays` and `MappedArrays` give small, self-contained examples showing how to create a new AbstractArray type.